bitcoin/src/consensus/encode.rs

```diff
@@ -350,7 +350,7 @@ pub trait Decodable: Sized {
     ///   avoid creating redundant `Take` wrappers. Failure to do so might result only in a tiny
     ///   performance hit.
     #[inline]
-    fn consensus_decode_from_finite_reader<R: BufRead + ?Sized>(
+    fn consensus_decode_from_finite_reader<R: Read + ?Sized>(
```

Comment:
I've just realized this is strictly a breaking change since people who started to rely on it in their own implementations would get broken by it.

Reply:
If we ship 0.33 then I'd also do the breaking `bitcoin-io` change. But I still prefer to have the `unbuffered` methods.

Reply:
I think it's ok to break the API in this way. Agreed that nobody has implemented the `Decodable` trait themselves.

Reply:
Maybe we should've made them sealed. :(